main.go

```diff
@@ -5,58 +5,106 @@ import (
+	token  = "token"
+)
+
 var (
-	handlerConfig = HandlerConfig{
+	handlerConfig = Config{
 		PluginConfig: sensu.PluginConfig{
-			Name:     "{{ .Name }}",
-			Short:    "{{ .Description }}",
+			Name:     "sensu-wavefront-handler",
+			Short:    "a wavefront metrics handler built for use with sensu",
 			Timeout:  10,
-			Keyspace: "sensu.io/plugins/{{ .Name }}/config",
+			Keyspace: "sensu.io/plugins/sensu-wavefront-handler/config",
 		},
 	}
 
-	handlerConfigOptions = ConfigOptions{
-		Example: sensu.PluginConfigOption{
-			Path:      "example",
-			Env:       "HANDLER_EXAMPLE",
-			Argument:  "example",
-			Shorthand: "e",
+	opts = []*sensu.PluginConfigOption{
+		&sensu.PluginConfigOption{
+			Path:      server,
+			Env:       "WAVEFRONT_SERVER",
+			Argument:  server,
+			Shorthand: "s",
 			Default:   "",
-			Usage:     "An example configuration option",
-			Value:     &handlerConfig.Example,
+			Usage:     "the address of the wavefront server",
+			Value:     &handlerConfig.Server,
+		},
+		&sensu.PluginConfigOption{
+			Path:      token,
+			Env:       "WAVEFRONT_TOKEN",
+			Argument:  token,
+			Shorthand: "t",
+			Default:   "",
+			Usage:     "the API token for the wavefront server",
+			Value:     &handlerConfig.Token,
 		},
-	}
-
-	options = []*sensu.PluginConfigOption{
-		&handlerConfigOptions.Example,
 	}
 )
 
 func main() {
-	handler := sensu.NewGoHandler(&handlerConfig.PluginConfig, options, checkArgs, executeHandler)
+	handler := sensu.NewGoHandler(&handlerConfig.PluginConfig, opts, checkArgs, executeHandler)
 	handler.Execute()
 }
 
-func checkArgs(_ *types.Event) error {
-	if len(handlerConfig.Example) == 0 {
-		return fmt.Errorf("--example or HANDLER_EXAMPLE environment variable is required")
+func checkArgs(event *corev2.Event) error {
+	if handlerConfig.Server == "" {
+		return fmt.Errorf("--server or WAVEFRONT_SERVER environment variable is required")
+	}
+	if handlerConfig.Token == "" {
+		return fmt.Errorf("--token or WAVEFRONT_TOKEN environment variable is required")
+	}
+	if !event.HasMetrics() {
+		return fmt.Errorf("event does not contain metrics")
 	}
 	return nil
 }
 
-func executeHandler(event *types.Event) error {
-	log.Println("executing handler with --example", handlerConfig.Example)
+func executeHandler(event *corev2.Event) error {
+	if len(event.Metrics.Points) == 0 {
+		log.Println("event does not contain metric points")
+		return nil
+	}
+
+	directCfg := &wavefront.DirectConfiguration{
+		Server:               handlerConfig.Server,
+		Token:                handlerConfig.Token,
+		BatchSize:            10000,
+		MaxBufferSize:        50000,
+		FlushIntervalSeconds: 1,
+	}
+
+	sender, err := wavefront.NewDirectSender(directCfg)
+	if err != nil {
+		return err
+	}
+
+	for _, point := range event.Metrics.Points {
+		tags := make(map[string]string)
+		for _, tag := range point.Tags {
+			tags[tag.Name] = tag.Value
+		}
+		sender.SendMetric(point.Name, point.Value, point.Timestamp, event.Entity.Name, tags)
```

Comment:
For Wavefront, are there restrictions on the format of the metric? For example, can tag name/value have spaces in them? Can `event.Entity.Name` have spaces in it?

I know that OpenTSDB is pretty strict about this, so we ended up adding an option to the handler to automatically convert spaces such that, if you have tags in your `Metric.Point` with spaces, they get transformed to something else. Same thing with the entity name.

For example, with an incoming `MetricTag` with `Name: "foo bar"` and `Value: "foo bar"`, that's automatically transformed into `foo_bar=foo_bar` or `foo-bar=foo-bar` before being sent to OpenTSDB.

Reply:
There are restrictions on the [wavefront data format](https://docs.wavefront.com/wavefront_data_format.html) but sanitation occurs in the go sdk so no additional validation is needed. If there are spaces in fields that are not allowed (ex. tag key, source/entity name, metric name) then the Wavefront sender automatically replaces those spaces with `-`.  See https://github.com/wavefrontHQ/wavefront-sdk-go/blob/master/senders/formatter.go#L357.